src/main/java/nl/mtvehicles/core/events/VehicleEntityEvent.java

```diff
@@ -0,0 +1,28 @@
+package nl.mtvehicles.core.events;
+
+import nl.mtvehicles.core.events.interfaces.HasVehicle;
+import nl.mtvehicles.core.infrastructure.models.MTVEvent;
+import nl.mtvehicles.core.infrastructure.models.Vehicle;
+import nl.mtvehicles.core.infrastructure.models.VehicleUtils;
+import org.bukkit.event.Cancellable;
+
+public class VehicleEntityEvent extends MTVEvent implements Cancellable, HasVehicle {
```

Comment:
When is this event called? The name is to generic.

Reply:
Split into more events! ;)